crates/subspace-service/src/sync_from_dsn/raw_block_import.rs

```diff
@@ -0,0 +1,93 @@
+use parity_scale_codec::Encode;
+use sc_client_api::{backend, BlockBackend, LockImportRun, ProofProvider};
+use sc_consensus::{BlockImportParams, ForkChoiceStrategy, StateAction};
+use sc_service::{ClientExt, Error};
+use sp_api::ProvideRuntimeApi;
+use sp_api::__private::BlockT;
+use sp_blockchain::HeaderBackend;
+use sp_consensus::BlockOrigin;
+use sp_consensus_subspace::{FarmerPublicKey, SubspaceApi};
+use sp_objects::ObjectsApi;
+use sp_runtime::traits::Header;
+use sp_runtime::Justifications;
+use tracing::{debug, error};
+
+pub type BlockWeight = u128;
+
+/// Write the cumulative chain-weight of a block ot aux storage.
+fn write_block_weight<H: Encode, F, R>(block_hash: H, block_weight: BlockWeight, write_aux: F) -> R
+where
+    F: FnOnce(&[(Vec<u8>, &[u8])]) -> R,
+{
+    let key = block_weight_key(block_hash);
+    block_weight.using_encoded(|s| write_aux(&[(key, s)]))
+}
+
+/// The aux storage key used to store the block weight of the given block hash.
+fn block_weight_key<H: Encode>(block_hash: H) -> Vec<u8> {
+    (b"block_weight", block_hash).encode()
```

Comment:
What block weight is and how it works was private in `sc-consensus-subspace` and is supposed to remain private there. Adding such code into `subspace-service` breaks abstractions in annoying way and can result in diverging code with non-trivial bugs. This whole module should probably be in `sc-consensus-subspace` instead because of this.

Reply:
Actually, the same `block_weight_key` and `write_block_weight` are already present in `sc-consensus-subspace` - I will remove the duplicates and export `write_block_weight` from the crate.

Reply:
That is exactly my point. They are already there and they are private for a reason. They shouldn't be made public to preserve implementation incapsulation. `import_raw_block` should be moved into `sc-consensus-subspace` instead. Alternatively `sc-consensus-subspace` should be modified to be ready for missing block weight entries since you initialize it to meaningless `0` value anyway.

Reply:
Yeah, after thinking about it some more, I think we don't need this helper function. Let's make consensus support missing weight data and inline contents of this function where it is used (my understanding is that there will be a single place).

Reply:
Did you mean treating an error as zero here? : https://github.com/subspace/subspace/blob/d159452bf4cd480837770d83c4f0cab3a28da048/crates/sc-consensus-subspace/src/block_import.rs#L661

Reply:
Check PR I linked above: https://github.com/subspace/subspace/pull/2767

Reply:
The suggested PR takes multiple testing and review rounds and doesn't seem as easy as you initially expected. Let's continue with this PR and return to the new one after we'll get the whole solution ready.

Reply:
It is quite easy actually. It changes nothing for the existing flow and introduces a special case handling for situation that occurs in fast sync. I don't see a reason to do a refactoring or to add hacks that we will have to remove shortly. It is just pure technical debt added with no benefit.